docs/developer/transform.rst

```diff
@@ -15,8 +15,87 @@ Two alternatives are provided:
 Both variants support:
 
 - Automatic broadcasting and alignment based on dimension labels.
-  This does also include sparse data, and operations mixing sparse and dense data are supported.
+  This does also include event data based on event lists, and operations mixing event lists and dense data are supported.
 - Automatic propagation of uncertainties, provided that the user-provided operation is built from known existing operations.
 - Operations between different data types, and operations that produce outputs with a new data type.
+- Handling of units.
 
-Note that ``transform`` and ``transform_in_place`` act only on the data elements and do **not** handle units.
+Basic usage
+-----------
+
+Transform requires:
+
+- A list of types (or type-combinations) to support.
+  Code is only instantiated for variables of those types.
+- A functor, typically a lambda or an "overloaded" lambda.
+  This can also be used to pass special flags, e.g., for disabling code generation for arguments that have variances, generating a runtime failure instead.
+
+Example 1
+~~~~~~~~~
+
+Transform with two variables with two type combinations:
+
+- ``a`` of type ``double`` and ``b`` of type ``float``
+- ``a`` of type ``double`` and ``b`` of type ``double``
+
+Since ``+`` is defined for ``units::Unit`` the same lambda can be used for data and unit.
```

Comment:
```suggestion
Since ``+`` is defined for ``units::Unit`` the same lambda can be used for data and unit.
This call to ``transform`` will add the two variables (or variable views) ``a`` and ``b`` and returns a new variable.
```